### DISCUSSION TITLE
As the finetune-training progressed, the colors of the generated images became weird

### DISCUSSION THREAD

**Discussion by A-pril** [huggingface/diffusers#8231] [Q&A] [answered]
Hello everyone, I just wanna to finetune the SD1.5 model alike "examples/text_to_image/train_text_to_image.py" does.  I have 25k images for finetune, but I just found the color of image will be weird after 10k steps.  I don't know whether it means that my code is wrong or the number of training steps mismatches the number of my training data? But the loss is still going down , and I don't think it is overfitting(or maybe it is :( 
here is the result in 4k steps
![image](https://github.com/huggingface/diffusers/assets/76250553/1f4103a3-86ef-4071-a6d7-600ab948aff4)
it's the result in 15k steps
![image](https://github.com/huggingface/diffusers/assets/76250553/3763de31-610c-42c5-9cbc-b8154b135516)

**Comment by tolgacangoz** [comment]
Hello @A-pril. Could you share the training command?

**Reply by A-pril** [reply]
Of course,here is the main part of the training config.

```
gradient_checkpointing: True
mixed_precision: "fp16"

pretrained_model_name_or_path: ../pre-trained/stable-diffusion-v1-5"
use_ema: True

resolution: 512
train_batch_size: 48
validation_samples: 16
validate_batch_size: 4 
gradient_accumulation_steps: 4
max_train_steps: 30000 
learning_rate: 1e-05 
max_grad_norm: 1 
validation_steps: 200 

lr_scheduler: "constant" 
lr_warmup_steps: 0 
prediction_type: "epsilon"
output_dir: "./exp" 
checkpoints_total_limit: 2
checkpointing_steps: 500
```

**Reply by A-pril** [reply]
> Hello @A-pril. Could you share the training command?

Hello, I just saw this discussion #7588 , It seems means that the resolution is very important.  And in my case,  I use 512 for SD finetune, but the 512*512 image is devided into 4 parts. So it's 4 256*256 images indeed. Maybe the resolution is too small?

**Reply by tolgacangoz** [reply]
[SD-1.5 was trained with 512x512 images.](https://huggingface.co/runwayml/stable-diffusion-v1-5) So, IMO it is not the reason. If results at the ~4k steps are what you expect, then stop early at that point?
`use_ema: False` can be tried?
Where is the 512x512 image divided into 4 parts?

**Reply by A-pril** [reply]
Thanks for your reply.The 512x512 image is composied by 4 256*256 images , it's just like a 2*2 grid.  Here is an example:
![image](https://github.com/huggingface/diffusers/assets/76250553/66c3e353-a40f-4233-9364-57ef981eb112)

The result in 4k steps is good, but I just wanna figure out the reason why the loss is going down but the color is abnormal. And according to some related papers, the best step is 10k.

**Reply by tolgacangoz** [reply]
[Merely loss might not be a good enough metric in this field.](https://discord.com/channels/879548962464493619/1009713274113245215/1165986567727362118)

**Reply by A-pril** [reply]
Thanks a lot!! It solves a big question in my heart. But I also have a question why the color become this kind style? Is it a common phenomenon? I have tried train very small data and the color will become weird either and even earlier.  So maybe it's related to the number of training data?  Looking forward to your reply, Thanks!

**Reply by bghira** [reply]
you might want to try training a zero-terminal snr / v-prediction model, as they represent the training data better than traditional epsilon with linear schedule

**Reply by tolgacangoz** [reply]
How do you define weirdness here, could you elaborate? Also, could you share more information about the datasets you are using?

**Reply by A-pril** [reply]
> you might want to try training a zero-terminal snr / v-prediction model, as they represent the training data better than traditional epsilon with linear schedule

Thank you, I have made a simple experiment with small data, but it seems not work well. I'll run the test on the full data set later.

**Reply by A-pril** [reply, marked as answer]
> How do you define weirdness here, could you elaborate? Also, could you share more information about the datasets you are using?

I just fix this this problem! I used the default scheduler for inference. Now I change to DDPM scheduler and the color is normal!! Really Thanks!!!
